### PR TITLE
Port CI changes for local building to release

### PR DESCRIPTION
**Pull request by ellismg**
This picks up

df20dd46ac86a45befcc39f4464ac97ea4b80074
9745abb3178486660093f5f80539b0a8e1746a43

from master and brings them into release so we do building of cross plat
distros on the local machine.

### CODE REVIEW

**Comment by ellismg** [comment]
@dotnet-bot test CI please

**Comment by ellismg** [comment]
@Priya91 Can you take a look?  This is porting some changes from master to release.

File: netci.groovy [hunk 1]
```diff
@@ -611,7 +303,7 @@ def supportedFullCyclePlatforms = ['Windows_NT', 'Ubuntu14.04', 'OSX']
                     else {
                         // Use Server GC for Ubuntu/OSX Debug PR build & test
                         def useServerGC = (configurationGroup == 'Release' && isPR) ? 'useServerGC' : ''
-                        shell("HOME=\$WORKSPACE/tempHome ./build.sh ${useServerGC} ${configurationGroup.toLowerCase()} /p:ConfigurationGroup=${configurationGroup} /p:WithoutCategories=IgnoreForCI")
+                        shell("HOME=\$WORKSPACE/tempHome ./build.sh ${useServerGC} ${configurationGroup.toLowerCase()} /p:TestNugetRuntimeId=${targetNugetRuntimeMap[os]} /p:ConfigurationGroup=${configurationGroup} /p:TestWithLocalLibraries=true /p:WithoutCategories=IgnoreForCI")
                         // Tar up the appropriate bits.  On OSX the tarring is a different syntax for exclusion.
```
Comment (Priya91) [inline netci.groovy]: Adding ```TestWithLocalLibraries=true``` on innerloop runs is undoing this [commit](https://github.com/dotnet/corefx/commit/2869dce82042f85f8d4fb51e059b35d8f2ea698e). We decided to run innerloop runs against release packages no..

**Comment by Priya91** [comment]
LGTM with comments ..

File: netci.groovy (hunk 1 above)
Comment (ellismg) [inline netci.groovy]: I forgot about that tweak, I will revert my changes there, thanks for pointing that out.

**Comment by ellismg** [comment]
@dotnet-bot test CI please

**Pull request merged**